tutorials/katacoda/thanos/2-lts/step2.md

```diff
@@ -0,0 +1,74 @@
+# Step 2 - Object Storage Configuration
+
+In this step, we will configure the object store and change sidecar to upload to the object-store.
+
+## Running Minio
+
+Now, execute the command
+
+```
+mkdir -p /storage/thanos
+```{{execute}}
+
+```
+docker run -d --name minio -v minio:/data -p 9000:9000 -e "MINIO_ACCESS_KEY=minio" -e "MINIO_SECRET_KEY=minio123" minio/minio:RELEASE.2019-01-31T00-31-19Z server /data
+```{{execute}}
+
+## Verification
+
+Now, you should have minio running well.
+
+To check if the Minio is working as intended, let's check out [here](https://[[HOST_SUBDOMAIN]]-9000-[[KATACODA_HOST]].environments.katacoda.com/minio/)
+
+## Configuration :
+
+The configuration file content :
+
+Click `Copy To Editor` for config to propagate the configs to the file `bucket_storage.yml`:
+
+<pre class="file" data-filename="bucket_storage.yml" data-target="replace">
+type: S3
+config:
+  bucket: "thanos"
+  endpoint: "minio:9000"
+  insecure: true
+  signature_version2: true
+  access_key: "minio"
+  secret_key: "minio123"
+</pre>
+
+Before moving forward, we need to stop the already running sidecar, and to do that follow the required steps as mentioned below:
+
+Run the command, here you will see all the running containers:
+
+```
+docker ps -a
+```{{execute}}
+
+We need to stop the `sidecar container` and we can do so by using the `CONTAINER ID`:
+
+`docker stop <CONTAINER ID>`
+
+**NOTE:** Replace `CONTAINER ID` with the ID of sidecar container.
+
+
+Now, execute the following command :
+
+```
+docker run -d --net=host --rm \
+    -v $(pwd)/bucket_storage.yml:/etc/prometheus/bucket_storage.yml \
+    -v $(pwd)/prometheus0_eu1_data:/prometheus \
+    --name sidecar \
+    -u root \
+    quay.io/thanos/thanos:v0.13.0 \
+    sidecar \
+    --tsdb.path                 /prometheus \
+    --objstore.config-file      /etc/prometheus/bucket_storage.yml \
+    --prometheus.url            http://127.0.0.1:9090 \
+    --http-address              0.0.0.0:19190    \
+    --grpc-address              0.0.0.0:19090 && echo "Store API exposed"
+```{{execute}}
```

Comment:
```Status: Downloaded newer image for quay.io/thanos/thanos:v0.12.2
level=info ts=2020-09-18T12:14:29.314332266Z caller=main.go:152 msg="Tracing will be disabled"
level=info ts=2020-09-18T12:14:29.315323453Z caller=options.go:23 protocol=gRPC msg="disabled TLS, keyand cert must be set to enable"
level=info ts=2020-09-18T12:14:29.31599092Z caller=factory.go:46 msg="loading bucket configuration"
level=error ts=2020-09-18T12:14:29.31675817Z caller=sidecar.go:312 err="WAL dir is not accessible. Is this dir a TSDB directory? If yes it is shared with TSDB?: stat /prometheus/wal: no such file or directory"
level=info ts=2020-09-18T12:14:29.317047487Z caller=sidecar.go:356 msg="starting sidecar"
level=info ts=2020-09-18T12:14:29.317500027Z caller=intrumentation.go:60 msg="changing probe status" status=healthy
level=info ts=2020-09-18T12:14:29.317940855Z caller=http.go:56 service=http/server component=sidecar msg="listening for requests and metrics" address=0.0.0.0:19190
level=info ts=2020-09-18T12:14:29.319108306Z caller=reloader.go:157 component=reloader msg="started watching config file and non-recursively rule dirs for changes" cfg= out= dirs=
level=info ts=2020-09-18T12:14:29.335330575Z caller=intrumentation.go:48 msg="changing probe status" status=ready
level=info ts=2020-09-18T12:14:29.335906222Z caller=grpc.go:106 service=gRPC/server component=sidecar msg="listening for StoreAPI gRPC" address=0.0.0.0:19090
level=info ts=2020-09-18T12:14:29.339649347Z caller=sidecar.go:220 msg="successfully loaded prometheusexternal labels" external_labels="{cluster=\"eu1\", replica=\"0\"}"
level=info ts=2020-09-18T12:14:29.340008875Z caller=intrumentation.go:48 msg="changing probe status" status=ready
```

There is no `WAL` file but Prometheus is generating one, and the above mentioned is the error generated after running this command. Is there any flag in Prometheus by which we can avoid this?

Reply:
So we might want to add step for creating `wal` directory. It's tedius but directory has to be there. But it can be empty (: 

Sorry! hidden knowledge!

Reply:
@bwplotka seems here the problem was not related to `wal` directory, I was using the wrong directory name to upload the data :( So sorry !!